rs-matter/src/data_model/root_endpoint.rs

```diff
@@ -56,15 +55,83 @@ pub const CLUSTERS: [Cluster<'static>; 10] = [
     group_key_management::CLUSTER,
 ];
 
-pub const fn endpoint(id: EndptId) -> Endpoint<'static> {
+/// The type of operational network (Ethernet, Wifi or (future) Thread)
+/// for which root endpoint meta-data is being requested
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum OperNwType {
+    Ethernet,
+    Wifi,
+}
+
+/// A utility function to create a root (Endpoint 0) object using the requested operational network type.
+pub const fn endpoint(id: EndptId, op_nw_type: OperNwType) -> Endpoint<'static> {
```

Comment:
The documentation mentions that this is supposed to be Endpoint 0 ... why do we pass in the endpoint id inside this method?

Could we document that or remove and auto-default to 0? Generally wondering if `endpoint` is the right name, would something like `create` be clearer (so I read `root_endpoint::create`)?

or maybe I should have `Endpoint::create_root` readability wise, however not sure how to achieve that.

Reply:
Only reason: I try to minimize the patch surface and keep the purpose of the patch to ideally one or a couple of goals.

It used to be named that way (even if maybe by me) - so my thinking was - let's keep the original name. We can have subsequent PRs that just rename stuff for readability. Otherwise I'm back into "here's this giant unreviewable PR"